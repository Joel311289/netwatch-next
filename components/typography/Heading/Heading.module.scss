.wrapper {
  color: currentColor;
  display: block;
  margin: 0;
}

.ellipsis {
  @include ellipsis;
}
