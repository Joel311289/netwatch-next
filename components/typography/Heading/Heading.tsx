import { createElement, HTMLAttributes } from 'react';
import cn from 'classnames';
import { ElementChildren, ElementHTML, ElementSkeleton } from '@/types';
import { withSkeleton } from '@/hoc/withSkeleton';
import styles from '@/components/typography/Heading/Heading.module.scss';

export const HeadingSize = {
  1: 'h1',
  2: 'h2',
  3: 'h3',
  4: 'h4',
  5: 'h5',
  6: 'h6'
};

export type HeadingProps = Partial<typeof defaultProps> &
  ElementHTML &
  ElementChildren<string> &
  ElementSkeleton & {
    level?: keyof typeof HeadingSize;
    ellipsis?: boolean;
  };

const defaultProps = {
  level: 2
};

const Heading = ({ className, children, level, ellipsis }: HeadingProps) => {
  const Element = ({ ...props }: HTMLAttributes<HTMLHeadingElement>) =>
    createElement(HeadingSize[level || 2], props, children);

  return (
    <Element className={cn(styles.wrapper, className, ellipsis && styles.ellipsis)}>
      {children}
    </Element>
  );
};

Heading.defaultProps = defaultProps;

const HeadingWithSkeleton = withSkeleton<HeadingProps>(Heading, { variant: 'heading' });

export default HeadingWithSkeleton;
