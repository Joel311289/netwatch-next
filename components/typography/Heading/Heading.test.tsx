import { render, screen } from '@testing-library/react';
import Heading, { HeadingSize } from '@/components/typography/Heading/Heading';

describe('Tests Heading component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Heading>Test</Heading>);

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(<Heading>Title</Heading>);

    expect(screen.getByText('Title')).toBeInTheDocument();
  });

  [1, 2, 3, 4, 5, 6].forEach((level) => {
    test(`renders component with level ${level} correctly`, () => {
      const { container } = render(
        <Heading level={level as keyof typeof HeadingSize}>Title</Heading>
      );

      expect(container).toContainHTML(`h${level}`);
    });
  });
});
