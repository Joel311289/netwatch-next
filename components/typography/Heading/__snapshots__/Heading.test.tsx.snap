// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Heading component renders component correctly 1`] = `
<div>
  <h2
    class="wrapper"
  >
    Test
  </h2>
</div>
`;
